PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/helpers/HDInsightHelperImpl.java

```diff
@@ -115,4 +119,10 @@ public static synchronized void initHDInsightLoader() {
             HDInsightLoader.setHHDInsightHelper(new com.microsoft.azuretools.azureexplorer.helpers.HDInsightHelperImpl());
         }
     }
+    
+    @NotNull
+    public NodeActionListener createAddNewHDInsightReaderClusterAction(@NotNull HDInsightRootModule module, @NotNull ClusterDetail clusterDetail) {
```

Comment:
Who calls this method?

Reply:
> Who calls this method?

When user right click on a Reader cluster node and selected "Link this cluster", this action will be called. Refer to [this line](https://github.com/microsoft/azure-tools-for-java/blob/73cd96ae4788c6b0580ccaf0073c83e883405970/Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/hdinsightnode/ClusterNode.java#L63)